### ISSUE TITLE
Language Models with Image Descriptors are Strong Few-Shot Video-Language Learners

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#195]
# TL; DR

- For video-language tasks, first generate lower-level natural language annotations and leverage the powerful few-shot capacity of language models to compose them into a temporally-coherent video description.

![image](https://user-images.githubusercontent.com/46308407/204274453-a22ed516-79cf-48f4-b11d-1e7c347ea336.png)

# Motivation

- Typical video-language models lack the generative capabilities and often struggle under few-shot setting (possibly due to domain gaps).

# Formulation

![image](https://user-images.githubusercontent.com/46308407/204274537-4f7175f2-2ac6-44bf-8787-e417755ce33c.png)

### Frame-level

- Use BLIP to generate captions for each sparsely sampled frame.

### Visual-token-level

- Prepare object vocabulary using OpenImage classes (~20k).
- Prepare event vocabulary using Visual Genome synsets and semantic role labeling.
- Prepare attribute vocabulary using Visual Genome attribute synset.

### Video-level

- Organize lower-level annotations using temporal templates (*first A, then B, finally C*).
- Three types of inputs to the GPT style language model: instruction + few-shot context + task query.
- Instruction: *"Generate a video caption based on the objects, events, attributes and frame captions. Example:"*
- Few-shot Context: *"Objects: First, bath toy. Then,..."* + *"Frame Captions: First, a toddler playing in a bathtub filled with toys. Then,..."*
- Task Query: *"Video Caption:"*
- Autoregressively decode the caption.

# Evaluation

- Employ three pretrained models: CLIP + BLIP + InstructGPT (+ AllenNLP for semantic role labeling).
- When choosing the in-context support examples, choose top-N based on the sentence similarity (i.e., relevance) using SentenceBERT and sort them in ascending order of similarity to compensate for the recency bias.

![image](https://user-images.githubusercontent.com/46308407/204275323-42e56f5a-01e9-4885-ac16-807a21b21b53.png)
![image](https://user-images.githubusercontent.com/46308407/204275339-43d72da7-d3af-4644-a494-776f89e15452.png)
![image](https://user-images.githubusercontent.com/46308407/204275350-a0572869-d434-4dd6-a806-3e590936e96b.png)
![image](https://user-images.githubusercontent.com/46308407/204275368-837873d5-330e-4273-ad0e-608a4a043952.png)
![image](https://user-images.githubusercontent.com/46308407/204275374-f0e8eb8e-1749-4559-bae1-1e4ab4d33adc.png)
![image](https://user-images.githubusercontent.com/46308407/204275391-5f948b88-eb23-4989-b3b9-74d8a5609edc.png)
![image](https://user-images.githubusercontent.com/46308407/204275403-b0c187e1-2439-439a-bf04-3fd5dea3cfb1.png)
![image](https://user-images.githubusercontent.com/46308407/204275413-b7d6706b-094b-462a-a3c9-b36af4bf2d69.png)

# Inspiration

- Clever work that compose multiple powerful pretrained models.
- These days, little inductive bias for data preparation and smart composition of foundation models may suffice for an interesting research paper.
- The whole work was done using 2 NVIDIA V100 GPUs.
- Possible improvements: the authors used global similarities computed by CLIP for object, event and attribute candidate selection. Regional vision-language parser will likely do better than naive CLIP. (or perhaps use open-set object detector)
- The way the authors orchestrated each model's strength into a complete framework is insightful. For example, their key idea is that language models are strong few-shot learners (thus, the title of GPT-3 paper), so they prepared all low level information in the form of natural language (not some abstract feature).
